src/core/base.jl

```diff
@@ -6,16 +6,58 @@ export
     run_generic_model, build_generic_model, solve_generic_model
 
 ""
-@compat abstract type AbstractPowerModel end
-
-""
 @compat abstract type AbstractPowerFormulation end
 
 ""
 @compat abstract type AbstractConicPowerFormulation <: AbstractPowerFormulation end
 
-""
-type GenericPowerModel{T<:AbstractPowerFormulation} <: AbstractPowerModel
+"""
+```
+type GenericPowerModel{T<:AbstractPowerFormulation}
+    model::JuMP.Model
+    data::Dict{String,Any} 
+    setting::Dict{String,Any}
+    solution::Dict{String,Any}
+    ref::Dict{Symbol,Any} # reference data
+end
+```
+where
+
+* `data` is the original data, usually from reading in a `.json` or `.m` (patpower) file
+* `setting` usually looks something like `Dict("output" => Dict("line_flows" => true))`, and
+* `ref` is a `Dict` with the following keys:
```

Comment:
If you prefer, I can change it to move the current description of `ref` to `build_ref()`, and use your comment in place of the description in `GenericPowerModel`.

It appears to me that `GenericPowerModel` is the exported "public-facing" interface, and is likely to be where users look up information on how to use `ref`. The docs were written with the consideration of letting them know the common keys that will be available, rather than having to be re-directed to look up the documentation of an internal function.

Anyhow, let me know what you decide? I'll make the changes accordingly.